app/models/device_token.rb

```diff
@@ -0,0 +1,16 @@
+class DeviceToken < ActiveRecord::Base
+  attr_accessible :token
+
+  def save
+    # Escape token: remove arrows and spaces
+    escaped_token = ""
+    token.each_char do |c|
+      escaped_token += c if c != " " && c != "<" && c != ">"
+    end
+    self.token = escaped_token
+
+    # Save if it does not already exist
+    return if DeviceToken.where("token = ?", self.token).count > 0
```

Comment:
This case cannot happen because each device has a uniq token. If a device sends a token twice in the same time, it is not a big deal. I am just saving one of them.